### PR TITLE
Search function using Autocomplete

### PR DESCRIPTION
**Pull request by vishal057**
I did Autocomplete search function using jquery and ajax. and I added this function in layout page. In controller I write method to call function.

### CODE REVIEW

File: ToDo_Assignment/Views/Shared/_Layout.cshtml [hunk 1]
```diff
@@ -46,6 +98,65 @@
...
+                        data: { "detail":  $("#selectedTaskId").val() },
+                        type: "POST",
+                        success: function (data) {
+                            document.getElementById("ID").value = data.id;
+                            document.getElementById("Task").value = data.task;
+                            document.getElementById("DescriptionModal").value = data.description;
+                            document.getElementById("AssignedTo").value = data.assignedTo;
+                            document.getElementById("DueDateModal").value = data.dueDate;
+                           
+
```
Comment (smartdusttechnologies) [inline ToDo_Assignment/Views/Shared/_Layout.cshtml]: remove these extra new lines.

File: ToDo_Assignment/Views/Shared/_Layout.cshtml [hunk 2]
```diff
@@ -34,6 +34,58 @@
...
+                </form>
+            </div>
+        </div>
+
+    </div>
+</div>  
+
+  
+
+
```
Comment (smartdusttechnologies) [inline ToDo_Assignment/Views/Shared/_Layout.cshtml]: remove extra new lines from here.

File: ToDo_Assignment/Controllers/HomeController.cs [hunk 3]
```diff
@@ -121,5 +121,31 @@ public IActionResult Error()
         {
             return View(new ErrorViewModel { RequestId = Activity.Current?.Id ?? HttpContext.TraceIdentifier });
         }
+
```
Comment (smartdusttechnologies) [inline ToDo_Assignment/Controllers/HomeController.cs:124]: Add proper comments to these Action method.

File: ToDo_Assignment/Controllers/HomeController.cs [hunk 4]
```diff
@@ -121,5 +121,31 @@ public IActionResult Error()
...
+            var toDoDal = new ToDoDal();
+            var tasks = toDoDal.get();
+            ViewData["id"] = detail;
+            var abc = tasks.Where(x => x.ID.Equals(detail)).FirstOrDefault();
+            return Json(abc);
+        }
+
+
+        [HttpGet]
+        public JsonResult GetSearchResults(string term)
```
Comment (smartdusttechnologies) [inline ToDo_Assignment/Controllers/HomeController.cs:147]: Add proper comments here as well.

File: ToDo.Dal/Operations/ToDoDal.cs [hunk 5]
```diff
@@ -10,7 +10,7 @@ namespace ToDo.Dal.Operations
 {
      public class ToDoDal
     {
-        private string sqlConnectionString = @"Data Source=DESKTOP-1GFB995;Initial Catalog=ToDoList;User ID=sa;Password=admin;Connect Timeout=30;Encrypt=False;TrustServerCertificate=False;ApplicationIntent=ReadWrite;MultiSubnetFailover=False";
```
Comment (smartdusttechnologies) [inline ToDo.Dal/Operations/ToDoDal.cs:13]: did you really needed to check-in this file. Seems unnecessary to me

**Review by smartdusttechnologies — COMMENTED** [review]
please complete all inline comments and commit again here.

File: ToDo_Assignment/Views/Shared/_Layout.cshtml (hunk 1 above)
Comment (vishal057) [reply ToDo_Assignment/Views/Shared/_Layout.cshtml]: Remove extra lines

File: ToDo_Assignment/Views/Shared/_Layout.cshtml (hunk 2 above)
Comment (vishal057) [reply ToDo_Assignment/Views/Shared/_Layout.cshtml]: Remove extra lines

File: ToDo_Assignment/Controllers/HomeController.cs (hunk 4 above)
Comment (vishal057) [reply ToDo_Assignment/Controllers/HomeController.cs:147]: comments the methods

File: ToDo.Dal/Operations/ToDoDal.cs (hunk 5 above)
Comment (vishal057) [reply ToDo.Dal/Operations/ToDoDal.cs:13]: Remove the changes

**Review by smartdusttechnologies — APPROVED** [review]
ok

**Review by smartdusttechnologies — APPROVED** [review]
ok

**Pull request merged**